lib/nrf_modem_lib/nrf91_sockets.c

```diff
@@ -810,11 +810,18 @@ static ssize_t nrf91_socket_offload_recvfrom(void *obj, void *buf, size_t len,
 		}
 
 		if (cliaddr->sa_family == NRF_AF_INET) {
-			nrf_to_z_ipv4(from, (struct nrf_sockaddr_in *)cliaddr);
+			struct sockaddr_in tmp_from;
+
+			nrf_to_z_ipv4((struct sockaddr *)&tmp_from,
+				      (struct nrf_sockaddr_in *)cliaddr);
+			memcpy(from, &tmp_from, MIN(*fromlen, sizeof(struct sockaddr_in)));
 			*fromlen = sizeof(struct sockaddr_in);
```

Comment:
I think that according to API, *fromlen shall be set to the number of bytes that were actually copied.
This is valid also for line 825

Reply:
linux is quite explicit on the behavior (return the full len, truncate array), while posix (https://pubs.opengroup.org/onlinepubs/007904875/functions/recvfrom.html) isn't explicit with what you should do with the fromlen in that scenario. @rlubos what is your understanding of this?

Reply:
I'd say to follow well-known Linux behavior in this case, as POSIX isn't very specific in this regard.

Reply:
My bad, you're right! Sorry for the confusion